app.js

```diff
@@ -0,0 +1,119 @@
+var createError = require('http-errors');
+var express = require('express');
+var app = express();
+const request = require('request');
+const zlib = require('zlib');
+const util = require('util');
+// const websiteUrlArr = ['https://www.zara.com/', 'https://www.sephora.fr/', 'https://www.stories.com/', 'https://www.nike.com/', 'https://www.adidas.com/', 'https://www.saksfifthavenue.com/', 'https://www.uniqlo.com/', 'https://www.cremedelamer.com/', 'http://www.calvinklein.com/'];
+const websiteUrlArr = ['https://www.zara.com/'];
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+
+app.get('/', (req, res) => {
+  res.json('Welcome to the express app')
+});
+
+app.get('/site-urls', async (req, res, next) => {
+
+  const scrapeRobotsTxtFiles = async (websiteUrlArr) => {
+    const siteMapUrlsObj = {};
+    for (websiteUrl of websiteUrlArr) {
+      robotTxtFileUrl = websiteUrl += 'robots.txt';
+      const urlArr = await requestRobotsTxtFiles(robotTxtFileUrl);
+      siteMapUrlsObj[websiteUrl] = urlArr;
+    }; 
+    return siteMapUrlsObj;
+  };
+
+  const gunzipPage = (body) => {
+    return new Promise((resolve, reject) => {
+      let nestedUrlArr = [];
+      zlib.gunzip(body, async (err, unzipped) => {
+        if (err) reject(err);
+        const unzippedSiteMap = unzipped.toString();
+        const space = /\s+/;
+        const unzippedSiteMapArr = unzippedSiteMap.split(space);
+        for (text of unzippedSiteMapArr) {
+          if (text.slice(-6) == '</loc>') {
+            const textLength = text.length;
+            const scrapedNextedSiteMapURl = text.slice(5, textLength - 6);
+            nestedUrlArr.push(scrapedNextedSiteMapURl);
+          }
+        };
+        resolve(nestedUrlArr);
+      });
+    })
+  };
+
+  const requestRobotsTxtFiles = (robotTxtFileUrls) => {
```

Comment:
I would export this functionality to another service file, something like: robots.service.js